src/hook.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "common.h"
+#include "posix.h"
+#include "repository.h"
+
+#include "git2/hook.h"
+
+#define GIT_HOOKS_DIR "hooks/"
+
+const char * const githooks[] = {
+	"applypatch-msg",
+	"pre-applypatch",
+	"post-applypatch",
+	"pre-commit",
+	"prepare-commit-msg",
+	"commit-msg",
+	"post-commit",
+	"pre-rebase",
+	"post-checkout",
+	"post-merge",
+	"pre-push",
+	"pre-receive",
+	"update",
+	"post-receive",
+	"post-update",
+	"push-to-checkout",
+	"pre-auto-gc",
+	"post-rewrite",
+};
+
+static int hook_dir(git_buf *out_dir, git_repository *repo)
+{
+	int err;
+	git_buf tmp_path = GIT_BUF_INIT;
+	git_buf cfg_path = GIT_BUF_INIT;
+	git_config *cfg;
+
+	assert(out_dir && repo);
+
+	/* We need to check for an override in the repo config */
+	err = git_repository_config__weakptr(&cfg, repo);
+	if (err != 0)
+		return err;
+
+	err = git_config_get_path(&cfg_path, cfg, "core.hooksPath");
+	if (err == GIT_ENOTFOUND) {
+		giterr_clear();
+		git_buf_puts(&tmp_path, git_repository_commondir(repo));
+		git_buf_joinpath(&tmp_path, tmp_path.ptr, GIT_HOOKS_DIR);
```

Comment:
Instead of doing `git_buf_puts`, `git_buf_joinpath`, you can just as well directly `git_buf_joinpath(&tmp_path, git_repository_commondir(repo), GIT_HOOKS_DIR)`.

Reply:
Agreed. But I can even use `git_repository_item_path` 😉.